runtime/lua/vim/lsp/util.lua

```diff
@@ -1722,10 +1783,11 @@ end
 
 --- Creates a `TextDocumentIdentifier` object for the current buffer.
 ---
+---@param bufnr (optional, number): Buffer handle, defaults to current
 ---@returns `TextDocumentIdentifier`
 ---@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocumentIdentifier
-function M.make_text_document_params()
-  return { uri = vim.uri_from_bufnr(0) }
+function M.make_text_document_params(bufnr)
```

Comment:
What do you think of using `win` as parameter here as well? I think the inconsistency could otherwise trip up users and lead to bugs.

(I'd generally prefer using bufnr everywhere, but I saw that's not possible because of `make_position_param` needing to get the cursor position)

Reply:
I feel like that would be more confusing personally, I see `make_position_params` as an inherently window-based method because it gets your position from the cursor, which is obviously tied to the window. This function not so much.

Reply:
> I see make_position_params as an inherently window-based method because it gets your position from the cursor, which is obviously tied to the window. This function not so much.

I think this is leaking implementation details to some degree. E.g. if you think about how you'd explain the difference:

"These functions generate parameters related to the document you're editing in the active window"

vs.

"These functions generate parameters for the document you're editing.  `make_x` takes a bufnr because it identifies the document, `make_y` takes winnr because it also needs the cursor position"